pkg/commands/patch/patch_modifier.go

```diff
@@ -137,12 +137,17 @@ func ModifiedPatchForLines(log *logrus.Entry, filename string, diffText string,
 }
 
 // I want to know, given a hunk, what line a given index is on
-func (hunk *PatchHunk) LineNumberOfLine(idx int) int {
-	lines := hunk.bodyLines[0 : idx-hunk.FirstLineIdx-1]
+func (hunk *PatchHunk) LineNumberOfLine(idx int) (int, error) {
+	n := idx - hunk.FirstLineIdx - 1
+	if n < 0 || len(hunk.bodyLines) <= n {
+		return -1, fmt.Errorf("line index out of range")
```

Comment:
what are your thoughts on having the line be zero if `n < 0` and the final line if `n >= len(hunt.bodyLines)` ?

Reply:
It sounds nice!!

I've changed it that way.